SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealObjectRef.h

```diff
@@ -72,7 +72,11 @@ struct SPATIALGDK_API FUnrealObjectRef
 	}
 
 	static UObject* ToObjectPtr(const FUnrealObjectRef& ObjectRef, USpatialPackageMapClient* PackageMap, bool& bOutUnresolved);
+	static void ToSoftObjectPtr(const FUnrealObjectRef& ObjectRef, FSoftObjectPtr& OutPtr);
```

Comment:
Why doesn't `ToSoftObjectPtr` return `FSoftObjectPtr`, but assigns it to the argument instead?

Reply:
FSoftObjectPtr can be "fat" since it can contain strings and such. I will check if it is properly movable and moved though.

Reply:
FSoftObjectPtr are not movable, so I'll keep it that way.

Reply:
Would Return-Value Optimisation take care of movement though?

Reply:
Maybe when returning from the function, but maybe not when assigning the value to the property.

Reply:
I mean, if we return from the function a temp value - it will automatically move as a part of RVO, so we don't need to bother with returning values via arguments.

Reply:
True, but only if we RVO into a brand new object. If we assign it to an already existing object, it will RVO into an ad-hoc temporary and copy itself into the object.
https://godbolt.org/z/fZz4Zi

Reply:
Got rid of the To/FromSoftObjectPtr in the end